client/src/screens/groups/components/GroupDetailsHeader.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { View, TouchableOpacity, Text } from 'react-native';
+import PropTypes from 'prop-types';
+
+import styles from './styles';
+
+import Icon from '../../../components/Icon';
+
+
+const GroupDetailsHeader = ({
+  groupIcon,
+  groupName,
+  tags,
+  memberAlready,
+  leaveGroup,
+  joinGroup,
+}) => (
+  <View style={styles.topContainer}>
+    <View style={styles.iconLeftHolder}>
+      <Icon style={styles.icon} name={groupIcon} size={30} />
+    </View>
+    <View style={styles.textContainer}>
+      <Text style={styles.titleTextBold} numberOfLines={1}>
+        {groupName}
+      </Text>
+      <Text style={styles.subtitleText} numberOfLines={1}>
+        {tags !== '' ? tags : 'No Tags'}
+      </Text>
+    </View>
+    <TouchableOpacity
```

Comment:
Sorry, I totally misunderstood what you where trying to do, Ill rewrite this to be more compact.

Reply:
@sdunster as in split this up even further?

Reply:
well yeah - make a component that is "IconButton" even if it is specific to users. Even better make it generic enough that we can re-use it across the app. Remember that special snowflakes aren't just bad for the code (extra work to maintain, need to have tests etc) but it's also bad for UI design (users don't recognise it as a button or don't know what it does because it's only used in this one spot in the app). If we don't have something else that would work better here lets make it a component that we can use in others places.